python/pyiceberg/catalog/dynamodb.py

```diff
@@ -0,0 +1,776 @@
+    _dict[DYNAMODB_COL_NAMESPACE]["S"] = to_database_name
+    _dict[DYNAMODB_COL_VERSION]["S"] = str(uuid.uuid4())
+    _dict[DYNAMODB_COL_UPDATED_AT]["N"] = current_timestamp_ms
+    return _dict
+
+
+def _get_create_database_item(database_name: str, properties: Properties) -> Dict[str, Any]:
+    current_timestamp_ms = str(round(time() * 1000))
+    _dict = {
+        DYNAMODB_COL_IDENTIFIER: {
+            "S": DYNAMODB_NAMESPACE,
+        },
+        DYNAMODB_COL_NAMESPACE: {
+            "S": database_name,
+        },
+        DYNAMODB_COL_VERSION: {
+            "S": str(uuid.uuid4()),
+        },
+        DYNAMODB_COL_CREATED_AT: {
+            "N": current_timestamp_ms,
+        },
+        DYNAMODB_COL_UPDATED_AT: {
+            "N": current_timestamp_ms,
+        },
+    }
+
+    for key, val in properties.items():
+        _dict[_add_property_prefix(key)] = {"S": val}
+
+    return _dict
+
+
+def _get_update_database_item(namespace_item: Dict[str, Any], updated_properties: Properties) -> Dict[str, Any]:
+    current_timestamp_ms = str(round(time() * 1000))
+
+    # noinspection PyDictCreation
+    _dict = {}
+    _dict[DYNAMODB_COL_IDENTIFIER] = namespace_item[DYNAMODB_COL_IDENTIFIER]
+    _dict[DYNAMODB_COL_NAMESPACE] = namespace_item[DYNAMODB_COL_NAMESPACE]
+    _dict[DYNAMODB_COL_VERSION] = {
+        "S": str(uuid.uuid4()),
+    }
+    _dict[DYNAMODB_COL_CREATED_AT] = namespace_item[DYNAMODB_COL_CREATED_AT]
+    _dict[DYNAMODB_COL_UPDATED_AT] = {
+        "N": current_timestamp_ms,
+    }
+
+    for key, val in updated_properties.items():
+        _dict[_add_property_prefix(key)] = {"S": val}
+
+    return _dict
+
+
+def _get_create_catalog_attribute_definitions() -> List[Dict[str, str]]:
+    return [
+        {
+            "AttributeName": DYNAMODB_COL_IDENTIFIER,
+            "AttributeType": "S",
+        },
+        {
+            "AttributeName": DYNAMODB_COL_NAMESPACE,
+            "AttributeType": "S",
+        },
+    ]
+
+
+def _get_key_schema() -> List[Dict[str, str]]:
+    return [
+        {
+            "AttributeName": DYNAMODB_COL_IDENTIFIER,
+            "KeyType": "HASH",
+        },
+        {
+            "AttributeName": DYNAMODB_COL_NAMESPACE,
+            "KeyType": "RANGE",
+        },
+    ]
+
+
+def _get_global_secondary_indexes() -> List[Dict[str, Any]]:
+    return [
+        {
+            "IndexName": DYNAMODB_NAMESPACE_GSI,
+            "KeySchema": [
+                {
+                    "AttributeName": DYNAMODB_COL_NAMESPACE,
+                    "KeyType": "HASH",
+                },
+                {
+                    "AttributeName": DYNAMODB_COL_IDENTIFIER,
+                    "KeyType": "RANGE",
+                },
+            ],
+            "Projection": {
+                "ProjectionType": "KEYS_ONLY",
+            },
+        }
+    ]
+
+
+def _get_namespace_properties(namespace_dict: Dict[str, str]) -> Properties:
+    namespace_properties = {}
```

Comment:
I'm a bit of a comprehension geek, but what do you think of:
```python
return [
    _remove_property_prefix(key): val
    for key, val in namespace_dict.items()
    if key.startswith(PROPERTY_KEY_PREFIX)
]
```

Reply:
Nice. Yeah looks much better. 
:)

Changed list comprehension to dict comprehension though in your code snippet.